tasks/insertionTasks.ts

```diff
@@ -0,0 +1,235 @@
+            `cd "${roslynRepoPath}" && roslyn-tools pr-finder -s "${startSHA}" -e "${endSHA}" --format changelog --label VSCode`,
+            { maxBuffer: 10 * 1024 * 1024 } // 10MB buffer
+        );
+        return stdout || '(failed to generate PR list, see pipeline for details)';
+    } catch (error) {
+        const errorMessage = error instanceof Error ? error.message : String(error);
+        logWarning(`PR finder failed, using empty list: ${errorMessage}`);
+        if (error instanceof Error && error.stack) {
+            console.log(`##[debug]${error.stack}`);
+        }
+        return '(failed to generate PR list, see pipeline for details)';
+    }
+}
+
+async function updatePackageJson(newVersion: string): Promise<void> {
+    console.log(`Updating package.json with Roslyn version ${newVersion}...`);
+    const packageJsonPath = 'package.json';
+    const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'));
+
+    if (!packageJson.defaults) {
+        throw new Error('Could not find defaults section in package.json');
+    }
+    packageJson.defaults.roslyn = newVersion;
+
+    fs.writeFileSync(packageJsonPath, JSON.stringify(packageJson, null, 2) + '\n');
+}
+
+async function updateChangelog(version: string, prList: string, buildNumber?: string, buildId?: string): Promise<void> {
+    console.log('Updating CHANGELOG.md...');
+    const changelogPath = 'CHANGELOG.md';
+    const text = fs.readFileSync(changelogPath, 'utf8');
+    const NL = os.EOL;
+
+    // Prepare PR list (filter out 'View Complete Diff' lines)
+    const prLines = prList
+        ? prList
+            .split(/\r?\n/)
+            .filter((l) => l.trim() && !l.includes('View Complete Diff'))
+            .map((line) => line.trim())
+        : [];
+
+    // Format PR list as sub-items with proper indentation
+    const formattedPRList = prLines.length > 0
+        ? prLines.map((line) => {
+            // If the line already starts with *, keep it, otherwise add it
+            const cleanLine = line.startsWith('*') ? line.substring(1).trim() : line;
+            return `  * ${cleanLine}`;
+          }).join(NL)
+        : '';
+
+    // Find the first top-level header "# ..."
+    const topHeaderRegex = /^(# .*?)(\r?\n|$)/m;
+    const headerMatch = topHeaderRegex.exec(text);
+    if (!headerMatch) {
+        throw new Error('CHANGELOG.md must contain at least one top-level header (#)');
+    }
+
+    const headerEndLineIndex = headerMatch.index + headerMatch[0].length;
+
+    // Prepare new Roslyn block
+    const prLink = buildNumber && buildId
+        ? `[#${buildNumber}](https://dev.azure.com/dnceng/internal/_build/results?buildId=${buildId})`
+        : '[#TBD](TBD)';
+
+    let newRoslynBlock = `* Bump Roslyn to ${version} (PR: ${prLink})`;
+
+    // Add PR list as sub-items if available and valid
+    const shouldSkipPRList = !prList || prList === '(failed to generate PR list, see pipeline for details)';
```

Comment:
I think we need to cleanup the handling of the `prList` a bit.  We're setting it to this complex failure string when it fails, but we don't even output that string anywhere. 

Couldn't we just make `prList` be nullable, and have the function that generates it return null when it fails?  And here we check if its null before adding the list to the changelog?